Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=f14413184b1de4dcbd5ec3e7c129c3ce2079f543
Commit:     f14413184b1de4dcbd5ec3e7c129c3ce2079f543
Parent:     0c7281c0faa1d0bdbdc647430cbdf7e0aed7f385
Author:     Matthew Garrett <mjg59@srcf.ucam.org>
AuthorDate: Wed Aug 20 14:08:57 2008 -0700
Committer:  Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Wed Aug 20 15:40:30 2008 -0700

    eeepc-laptop: fix use after free
    
    eeepc-laptop uses the hwmon struct after unregistering the device, causing
    an oops on module unload.  Flip the ordering to fix.
    
    Signed-off-by: Matthew Garrett <mjg@redhat.com>
    Cc: Henrique de Moraes Holschuh <hmh@hmh.eng.br>
    Cc: Corentin Chary <corentincj@iksaif.net>
    Cc: Karol Kozimor <sziwan@users.sourceforge.net>
    Cc: <stable@kernel.org>		[2.6.26.x]
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

 drivers/misc/eeepc-laptop.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/misc/eeepc-laptop.c b/drivers/misc/eeepc-laptop.c
index 9e8d79e..facdb98 100644
--- a/drivers/misc/eeepc-laptop.c
+++ b/drivers/misc/eeepc-laptop.c
@@ -553,9 +553,9 @@ static void eeepc_hwmon_exit(void)
 	hwmon = eeepc_hwmon_device;
 	if (!hwmon)
 		return ;
-	hwmon_device_unregister(hwmon);
 	sysfs_remove_group(&hwmon->kobj,
 			   &hwmon_attribute_group);
+	hwmon_device_unregister(hwmon);
 	eeepc_hwmon_device = NULL;
 }
 
